android/engine/src/main/java/org/smartregister/fhircore/engine/domain/model/FhirResourceConfigs.kt

```diff
@@ -90,7 +91,54 @@ data class ResourceConfig(
   val configRules: @RawValue List<RuleConfig>? = null,
   val planDefinitions: List<String>? = null,
   val attributesToUpdate: List<KeyValueConfig>? = emptyList()
-) : Parcelable, java.io.Serializable
+) : Parcelable, java.io.Serializable {
+  fun interpolate(computedValuesMap: Map<String, Any>) =
+    this.copy(
+      dataQueries = interpolateDataQueries(dataQueries = dataQueries, computedValuesMap),
+      relatedResources =
+        relatedResources.map {
+          it.copy(
+            dataQueries = interpolateDataQueries(dataQueries = dataQueries, computedValuesMap)
+          )
+        }
+    )
+
+  private fun interpolateDataQueries(
+    dataQueries: List<DataQuery>?,
+    computedValuesMap: Map<String, Any>
+  ): List<DataQuery>? {
+    return dataQueries?.map { dataQuery ->
+      dataQuery.copy(
+        filterCriteria =
+          dataQuery.filterCriteria.map { filterCriterionConfig ->
+            when (filterCriterionConfig.dataType) {
+              Enumerations.DataType.REFERENCE ->
+                (filterCriterionConfig as FilterCriterionConfig.ReferenceFilterCriterionConfig)
+                  .copy(value = filterCriterionConfig.value?.interpolate(computedValuesMap))
+              Enumerations.DataType.STRING ->
+                (filterCriterionConfig as FilterCriterionConfig.StringFilterCriterionConfig).copy(
+                  value = filterCriterionConfig.value?.interpolate(computedValuesMap)
+                )
+              Enumerations.DataType.URI ->
+                (filterCriterionConfig as FilterCriterionConfig.UriFilterCriterionConfig).copy(
+                  value = filterCriterionConfig.value?.interpolate(computedValuesMap)
+                )
+              Enumerations.DataType.CODE ->
+                (filterCriterionConfig as FilterCriterionConfig.TokenFilterCriterionConfig).copy(
+                  value =
+                    filterCriterionConfig.value?.copy(
+                      code = filterCriterionConfig.value?.code?.interpolate(computedValuesMap)
+                    )
+                )
```

Comment:
Can we use the `FilterCriterion` class in the when clause instead? Otherwise all the supported data types must be handled. For instance, the config `TokenFilterCriterionConfig` is used for CODE, CODEABLECONCEPT and CODING types.